src/Core/Grid/Search/Factory/DecoratedSearchCriteriaFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Search\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Interface DecoratedSearchCriteriaFactory defines contract for decorated search criteria factory
+ */
+interface DecoratedSearchCriteriaFactory
```

Comment:
I don't think we should try to override that comes from the user, but maybe we need it? I need to be convinced

Reply:
it does not override what comes from user, what it does is keeps exactly the same data except it extends criteria with dynamic `parent_category_id` value. :)

Reply:
yes but resolving the data inside SearchCriteria is not "their" responsibility, this responsibility is inside a Controller Argument Resolver